src/main/java/org/skriptlang/skript/bukkit/itemcomponents/ComponentWrapper.java

```diff
@@ -0,0 +1,150 @@
+package org.skriptlang.skript.bukkit.itemcomponents;
+
+import ch.njol.skript.util.ItemSource;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.converter.Converter;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * A wrapper that allows access and modification of a specific component from an {@link ItemStack}
+ * or a stand-alone component.
+ * @param <T> The type of component
+ */
+public abstract class ComponentWrapper<T> {
+
+	private final @Nullable ItemSource itemSource;
+	private final @Nullable ItemStack itemStack;
```

Comment:
It seems a bit duplicative to have this field and `itemSource`. Maybe a meta getter on `itemSource` could help?